src/MongoDB.Driver.Core/Core/Authentication/ScramShaAuthenticator.cs

```diff
@@ -30,11 +30,14 @@ namespace MongoDB.Driver.Core.Authentication
     /// </summary>
     public abstract class ScramShaAuthenticator : SaslAuthenticator
     {
+        // todo: this field has been renamed due to the following ambiguity error:
```

Comment:
Not sure how can we solve this issue. I suppose we cannot rename the method since it's public. 
As I see, it's the naming warning: https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1724-type-names-should-not-match-namespaces?view=vs-2019
Is it a good solution just to turn off this warning here?

Reply:
I disabled this warning in the last commit.